mlflow/evaluation.py

```diff
@@ -0,0 +1,224 @@
+from typing import Dict, Union
+import entrypoints
+import warnings
+import mlflow
+from mlflow.exceptions import MlflowException
+
+
+class EvaluationMetrics(dict):
+    pass
+
+
+class EvaluationArtifact:
+
+    @property
+    def content(self):
+        """
+        The content of the artifact (representation varies)
+        """
+        raise NotImplementedError()
+
+    @property
+    def location(self) -> str:
+        """
+        The location of the artifact
+        """
+        raise NotImplementedError()
+
+
+class EvaluationResult:
+
+    @classmethod
+    def load(cls, path):
+        """Load the evaluation results from the specified local filesystem path"""
+        raise NotImplementedError()
+
+    def save(self, path):
+        """Write the evaluation results to the specified local filesystem path"""
+        # We will likely avoid serializing artifacts themselves, just locations.
+        # Deserialization will resolve locations to artifact contents.
+        raise NotImplementedError()
+
+    @property
+    def metrics(self) -> EvaluationMetrics:
+        """
+        A dictionary mapping scalar metric names to scalar metric values
+        """
+        raise NotImplementedError()
+
+    @property
+    def artifacts(self) -> Dict[str, EvaluationArtifact]:
+        """
+        A dictionary mapping standardized artifact names (e.g. "roc_data") to
+        artifact content and location information
+        """
+        raise NotImplementedError()
```

Comment:
Can we provide a default implementation for each of these methods and a default constructor? This way, plugin implementers don't have to redefine `EvaluationResult` themselves unless they want to add new methods.